docs: added jvs docs (#114)
abcxyz/jvs: docs/jvs-apis.md

> # JVS Usage

Reviewer: Let's **not** mix API doc vs. quick setup doc.

API doc - focusing on the API spec. How to call the API and what's the response, etc.
Setup doc - how to setup JVS in a cloud project and sample interactions

Feel free to separate them in two PRs (I haven't added the lumberjack setup doc).

Author: done

# JVS APIs

**JVS is not an official Google product.**

## Justification API

### API Spec

> Currently, Justification API uses [gRPC](https://grpc.io/) and only supports the
> core functionality `Create Token` which creates a justification and generates a
> token. Justification API processes
> [CreateJustificationRequests](https://github.com/abcxyz/jvs/blob/main/protos/v0/jvs_request.proto#L23-L28)
> and provides signed tokens. See
> [JVSService](https://github.com/abcxyz/jvs/blame/e718d4664467b880991b8e2a400070c2aa93a0b9/blob/main/protos/v0/jvs_service.proto)
> for details.

Reviewer: ```suggestion
Justification API is a [gRPC](https://grpc.io/) service. It accepts
[CreateJustificationRequests](https://github.com/abcxyz/jvs/blob/main/protos/v0/jvs_request.proto#L23-L28)
and responses with signed justification tokens as JWTs. See
[JVSService](https://github.com/abcxyz/jvs/blame/e718d4664467b880991b8e2a400070c2aa93a0b9/blob/main/protos/v0/jvs_service.proto)
for details.
```

### Setup Knobs

> Currently, Justification API loads in the config with the env variables
> specified on the host. See

Reviewer: ```suggestion
Justification API loads configs from environment variables. See
```

[JustificationConfig](https://github.com/abcxyz/jvs/blob/main/pkg/config/justification_config.go#L32-L49)
for details of supported config env variables.

## Public Key API

### API Spec

Public Key API exposes a JWKS endpoint which is found at
`${PUBLIC_KEY_SERVER_URL}/.well-known/jwks`. This endpoint will contain the JWK
used to verify all Auth0-issued JWTs. Refer to
[JWKs](https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-key-sets).

### Setup Knobs

> Currently, Public Key API loads in the config with the env variables specified

Reviewer: ```suggestion
Public Key API loads configs from environment variables.
```

on the host. See
[PublicKeyConfig](https://github.com/abcxyz/jvs/blob/main/pkg/config/public_key_config.go#L26-L35)
for details of supported config env variables.

## Cert Rotation API

### API Spec

> Cert Rotation API should be triggerd by cron job e.g.
> [Cloud Scheduler](https://cloud.google.com/scheduler). To support key rotation,
> Cert Rotation API needs to generate a new Key Version, move applications to the
> new Key Version, and finally disable the old version.

Reviewer: ```suggestion
Cert Rotation API will do the following based on multiple conditions (link to code):
  * Create new key verions
  * Set the new primary key version
  * Disable or delete old key versions
  
The service is meant to be triggered by [Cloud Scheduler](https://cloud.google.com/scheduler) job.
```

Author: done

### Setup Knobs

> Currently, Cert Rotation API loads in the config with the env variables
> specified on the host. See

Reviewer: ```suggestion
Cert Rotation API loads configs from environment variables. See
```

> # JVS APIs

Reviewer: As we discussed offline, let's make it a API spec doc.

For Justification API, we can reference the grpc service definition at https://github.com/abcxyz/jvs/blob/main/protos/v0/jvs_service.proto

For public key API, we can reference JWKs definition https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-key-sets

For cert rotate API, AFAIK it's empty http call (triggered by cloud scheduler). We should describe its behavior.

Reviewer: It's OK if you want to split it up to be per-API. We can do the following for each API:

1. Spec
2. Setup knobs (env vars, flags, etc.)

Author: done
